### PR TITLE
Fix ports in L4NetLB

### PR DESCRIPTION
**Pull request by kl52752**
In L4NetLB NodePort is used by namer to create BackendService name.
Previously NodePort was not set and by default its value was 0 so
all backends had the same name and we there were error in creating
multiple services
To test NodePort set multi-services test were written

Change port ranges calculation for creating forwarding rule.
Previously we assigned first continue port Range from service,
now it is calculated from service assigned ports to
create boundary range MINPORT-MAXPORT.

### CODE REVIEW

**Comment by kl52752** [comment]
\assign @freehan

File: pkg/loadbalancers/l4netlb.go [hunk 1]
```diff
@@ -79,8 +79,10 @@ func NewL4NetLB(service *corev1.Service, cloud *gce.Cloud, scope meta.KeyType, n
 		backendPool:         backends.NewPool(cloud, namer),
 	}
 	portId := utils.ServicePortID{Service: l4netlb.NamespacedName}
-	l4netlb.ServicePort = utils.ServicePort{ID: portId,
+	l4netlb.ServicePort = utils.ServicePort{
+		ID:           portId,
 		BackendNamer: l4netlb.namer,
+		NodePort:     int64(service.Spec.Ports[0].NodePort),
```
Comment (prameshj) [inline pkg/loadbalancers/l4netlb.go:85]: We should only set this if the value is non-zero. It is possible to disable NodePort allocation for LoadBalancer services using https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1864-disable-lb-node-ports. 

Also, what is this used for? To annotate the instance group with the service port?

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -612,6 +612,29 @@ func GetPortsAndProtocol(svcPorts []api_v1.ServicePort) (ports []string, portRan
...
+			minPort = svcPort.Port
+		}
+		if svcPort.Port > maxPort {
+			maxPort = svcPort.Port
+		}
+	}
+	return minPort, maxPort
+}
+
+func GetPortMinMaxRangeAndProtocol(svcPorts []api_v1.ServicePort) (portRange, protocol string) {
```
Comment (prameshj) [inline pkg/utils/utils.go:629]: test for this util?

Also, nit: You can drop the Get prefix and call this MinMaxPortRangeAndProtocol.

File: pkg/l4netlb/l4netlbcontroller_test.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (prameshj) [inline pkg/l4netlb/l4netlbcontroller_test.go:2]: nit: should this test addition be part of a separate commit/PR? It is not adding a test for the multiple port forwarding rule right?

I think this test should also verify forwarding rule(in addition to backend service) and have a testcase for multiple ports forwarding rule.

File: pkg/loadbalancers/l4netlb.go (hunk 1 above)
Comment (kl52752) [reply pkg/loadbalancers/l4netlb.go:85]: Node porst is used by Namer to create backend service name `sp.BackendNamer.IGBackend(sp.NodePort)`

File: pkg/utils/utils.go (hunk 2 above)
Comment (kl52752) [reply pkg/utils/utils.go]: done + test added

File: pkg/l4netlb/l4netlbcontroller_test.go (hunk 3 above)
Comment (kl52752) [reply pkg/l4netlb/l4netlbcontroller_test.go:2]: I added here test for multiservice because this way I can test changes with Node Port where we calculate new name for backend based on NodePort. I will add test for multiport forwarding rule.

Comment (kl52752) [reply pkg/l4netlb/l4netlbcontroller_test.go:2]: done added test for multiple ports forwarding rule

File: pkg/loadbalancers/l4netlb.go (hunk 1 above)
Comment (kl52752) [reply pkg/loadbalancers/l4netlb.go:85]: As I can see it is also used in L7 LB
